algo/src/main/java/com/hartwig/serve/sources/ckb/CkbExtractor.java

```diff
@@ -85,255 +82,213 @@ public CkbExtractor(@NotNull Knowledgebase source, @NotNull final EventExtractor
+                entries.parallelStream().map(this::getExtractionResult).peek(e -> tracker.update()).filter(Objects::nonNull)
+                .collect(Collectors.toList());
 
         return ExtractionFunctions.merge(extractions);
     }
 
-    @NotNull
-    private ExtractionResult getExtractionResult(CkbEntry entry, String gene, String event, List<ExtractionResult> extractions,
-            Variant variant) {
-        EventExtractorOutput extractionOutput = eventExtractor.extract(gene, null, entry.type(), event);
-        String sourceEvent;
-        if (!gene.equals(CkbConstants.NO_GENE)) {
-            sourceEvent = gene + " " + event;
-        } else {
-            sourceEvent = event;
+    @Nullable
+    private ExtractionResult getExtractionResult(@NotNull CkbEntry entry) {
+        if (entry.variants().isEmpty()) {
+            throw new IllegalStateException("A CKB entry without variants has been provided for extraction: " + entry);
         }
+        int variantCount = entry.variants().size();
+        Variant variant = entry.variants().get(0);
+        String event = variantCount > 1 ? concat(entry.variants()) : CkbEventAndGeneExtractor.extractEvent(variant);
+        String gene = variantCount > 1 ? "Multiple" : CkbEventAndGeneExtractor.extractGene(variant);
 
-        Set<ActionableEntry> actionableEntries = actionableEntryFactory.create(entry, sourceEvent, gene);
-
-        EventInterpretation interpretation = ImmutableEventInterpretation.builder()
-                .source(source)
-                .sourceEvent(sourceEvent)
-                .interpretedGene(gene)
-                .interpretedEvent(event)
-                .interpretedEventType(entry.type())
-                .build();
-
-        ExtractionResult extraction = toExtractionResult(event, gene, null, extractionOutput, actionableEntries, interpretation);
-
-        if (generateKnownEvents) {
-            extractions.add(CkbVariantAnnotator.annotate(extraction, variant));
+        if (entry.type() == EventType.UNKNOWN) {
+            LOGGER.warn("No event type known for '{}' on '{}'", event, gene);
+            return null;
         } else {
-            extractions.add(extraction);
-        }
+            EventExtractorOutput extractionOutput = eventExtractor.extract(gene, null, entry.type(), event);
+            String sourceEvent = gene.equals(CkbConstants.NO_GENE) ? event : gene + " " + event;
+
+            Set<ActionableEntry> actionableEntries = actionableEntryFactory.create(entry, sourceEvent, gene);
+
+            EventInterpretation interpretation = ImmutableEventInterpretation.builder()
+                    .source(source)
+                    .sourceEvent(sourceEvent)
+                    .interpretedGene(gene)
+                    .interpretedEvent(event)
+                    .interpretedEventType(entry.type())
+                    .build();
+
+            List<CodonAnnotation> codons = curateCodons(extractionOutput.codons());
```

Comment:
Unrelated to your change, but maybe `curateCodons` should take in and return an `ExtractionOutput` such that we don't have to pass these specific codons around anymore.

Also, FYI: We curate codon 600 of BRAF because at that time the 38 ensembl cache used the wrong transcript for BRAF. I assume this is fixed by now (otherwise we'd have problems using OncoPanel 38!), we'd know for sure if we run a BRAF V600E through PAVE on 38 (or alternatively, run COLO pipeline on 38 since COLO has V600E mutation)